src/orion/algo/random.py

```diff
@@ -39,7 +35,7 @@ def seed_rng(self, seed):
     @property
     def state_dict(self):
         """Return a state dict that can be used to reset the state of the algorithm."""
-        return {'rng_state': self.rng.get_state(), 'trial_info': self.trial_info}
+        return {'rng_state': self.rng.get_state()}
```

Comment:
If we remove abstractmethod from `observe` to make its definition optional in children class, we could remove it altogether in `Random` and reuse super().state_dict/set_state to insert `trial_info` in the states. This would solve the issue for the call on `super` in PrimaryAlgo.

Reply:
Even we remove abstractmethod of observe, but not sure how many actual algorithm will stop implement their own. This will remove the super call in PrimaryAlgo.observe if the child does not need their own observe. As for state_dict and set_state, are you saying we ask user to explicitly in their concrete child algorithm? 

```
def state_dict(self):
      child_state_dict = xxx
      child_state_dict.update(super.state_dict)
def set_state(self, state_dict):
      super.set_state(state_dict)
```

And even for observe, even the child algorithm get its own algorithm, we also ask them to call super.observe like this?
```
def observe(self, points, results):
       super.observe(points, results)
      # child observer code here ...
```

This way it indeed simplify the code in PrimaryAlgo, but actually ask sub-algorithm to call super explicitly, I guess people could make mistake even we can document about this.

Reply:
Yes I meant the latter. It may well be an common mistake to forget it, but if we include the call to super in the cookie-cutter template with a clear documentation of this call then the mistake should happen rarely actually.

Reply:
okay, using the template could be a way, normally I would say split such a feature to depend part on orion own code and part on user's code could be risky, but let's get it work first.